baseapp/abci.go

```diff
@@ -38,6 +38,9 @@ const (
 )
 
 func (app *BaseApp) InitChain(req *abci.RequestInitChain) (*abci.ResponseInitChain, error) {
+	app.mtx.Lock()
+	defer app.mtx.Unlock()
+
 	if req.ChainId != app.chainID {
 		return nil, fmt.Errorf("invalid chain-id on InitChain; expected: %s, got: %s", app.chainID, req.ChainId)
 	}
```

Comment:
For my understanding, why did we leave out these [lines](https://github.com/dydxprotocol/cosmos-sdk/blob/bdf96fddf421e90574e53af383c1468fdb33d7a3/baseapp/abci.go#L74-L82) (back-ported [here](https://github.com/dydxprotocol/cosmos-sdk/commit/ae5ce197934ccb21acd6dba43d875ecb77c27b2f))?